include/libp2p/basic/read_full.hpp

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LIBP2P_BASIC_READ_FULL_HPP
+#define LIBP2P_BASIC_READ_FULL_HPP
+
+#include <libp2p/basic/reader.hpp>
+#include <libp2p/common/span_size.hpp>
+#include <memory>
+
+namespace libp2p {
+  inline void _readFull(const std::shared_ptr<basic::Reader> &reader,
+                        gsl::span<uint8_t> out,
+                        basic::Reader::ReadCallbackFunc &&cb, size_t total) {
+    if (out.empty()) {
+      throw std::logic_error{"libp2p::readFull 0 bytes requested"};
```

Comment:
I think it's non-breaking case. I think it would be enough to insert boost assert here